### PR TITLE
SUP-820 Pull latest 1.4 version terraform from docker image

### PR DESCRIPTION
**Pull request by jradtilbrook**
Update the terraform version used in CI to 1.4

This is implemented through multistage build for caching performance

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,9 +5,4 @@ RUN apt-get update \
     && go install github.com/mfridman/tparse@latest \
     && go install github.com/lox/buildkite-test-analytics-go@latest
 
-ENV TERRAFORM_VERSION=0.14.11
```
Comment (yob) [inline Dockerfile:8]: wowee, ye olde version

**Comment by jradtilbrook** [comment]
@mcncl how do you think we should implement this? In the multistage build I've done now or using the existing method to download binary explicitly?

I think if we do it the multistage way, we could use the pull-through cache again. @yob could you weigh in as well?

File: Dockerfile (hunk 1 above)
Comment (jradtilbrook) [reply Dockerfile:8]: Very!!

**Comment by mcncl** [comment]
I use TF 1+ for everything and never use this Dockerfile, so it's reasonably safe to assume we can bump this to the latest stable

**Comment by jradtilbrook** [comment]
I'm very surprised to see this works straight away. Maybe the other recent upgrades we've done has had an impact

**Comment by mcncl** [comment]
@jradtilbrook it's always Just Worked™ with 1+ for me

File: Dockerfile [hunk 2]
```diff
@@ -5,9 +5,4 @@ RUN apt-get update \
...
-
-RUN cd /tmp && \
-    curl -O https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
-    unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
-    cp terraform /usr/local/bin/terraform
+COPY --from=hashicorp/terraform:1.4 /bin/terraform /usr/local/bin/terraform
```
Comment (jradtilbrook) [inline Dockerfile:8]: @mcncl @yob are you okay with this approach?

An added benefit would be that dependabot could help us keep up to date with v1.5+ as well

**Review by mcncl — APPROVED** [review]

**Pull request merged**